src/main/java/seedu/address/ui/BrowserPanel.java

```diff
@@ -140,18 +140,18 @@ private void handleSucceeded() {
         isLoadingReaderView = false;
 
         // Load reader view if reader view mode is selected but not loaded
-        if (viewMode.getViewType().equals(ViewType.READER)
+        try {
+            URL url = new URL(webEngine.getLocation());
+            if (viewMode.getViewType().equals(ViewType.READER)
                 && !hasLoadedReaderView
-                && currentLocation.equals(webEngine.getLocation())) {
-            try {
-                URL url = new URL(webEngine.getLocation());
+                && url.equals(currentLocationUrl)) {
```

Comment:
Note we use URL#equals to compare equality of URLs. This is probably more correct than what we were doing before.